### PR TITLE
The entropy gain calculation, working perfectly

### PR DESCRIPTION
**Pull request by matteobecchi**
Requested Reviewers: @andrewtarzia

### CODE REVIEW

File: src/dynsight/_internal/analysis/shannon_entropy.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    n_bins: int,
+) -> float:
+    """Compute the entropy of a data distribution.
+
+    It is normalized so that a uniform distribution has unitary entropy.
+
+    * Original author: Matteo Becchi
+
+    Parameters:
+        data : the dataset of which the entropy has to be computed.
```
Comment (andrewtarzia) [inline src/dynsight/_internal/analysis/shannon_entropy.py:21]: Change to formatting like this:

```python
        data:
            COMMENT.
```

Removing the typing information.

**Comment by andrewtarzia** [comment]
The analysis module needs to be added to the documentation - can you do that @matteobecchi

File: src/dynsight/_internal/analysis/shannon_entropy.py (hunk 1 above)
Comment (matteobecchi) [reply src/dynsight/_internal/analysis/shannon_entropy.py]: Done

**Comment by matteobecchi** [comment]
I didn't added the documentation for the analysis module because Simone did (and I was expecting his PR to go trough before mine).

**Comment by andrewtarzia** [comment]
Ok that makes sense, thank you.

**Comment by matteobecchi** [comment]
I wrote the (minimal) documentation for the `analysis` module.

**Review by andrewtarzia — APPROVED** [review]

**Pull request merged**